bfgs.go

```diff
@@ -159,3 +159,7 @@ func (b *BFGS) NextDirection(loc *Location, dir []float64) (stepSize float64) {
 	floats.Scale(-1, dir)
 	return 1
 }
+
+func (_ BFGS) Requires() (grad, hess bool) {
```

Comment:
You don't need the underscore, just func(BFGS). It should be func(*BFGS) though to be consistent. This change goes for the other optimizers.